src/Aspire.Hosting.Azure/Utils/BicepIdentifierHelpers.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Aspire.Hosting.Azure.Utils;
+
+internal static partial class BicepIdentifierHelpers
+{
+    // See rules from Bicep's highlightjs implementation:
+    // https://github.com/Azure/bicep/blob/a992bdf2d4d7c5c7dec684b7d0de4db9cb260f8a/src/highlightjs/src/bicep.ts#L12
+    [GeneratedRegex("^[A-Za-z_][A-Za-z0-9_]*$")]
```

Comment:
Note that this regex changed to no longer allow the Kelvin Sign (which a-z, ignore case allows).